adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs

```diff
@@ -105,8 +105,7 @@ public string GetAuthorityHost()
             {
                 var authorityUri = new Uri(this.Authority);
                 var host = authorityUri.Host;
-                string path = authorityUri.AbsolutePath.Substring(1);
-                string tenant = path.Substring(0, path.IndexOf("/", StringComparison.Ordinal));
+                string tenant = authorityUri.Segments[authorityUri.Segments.Length - 1].TrimEnd('/');
```

Comment:
>authorityUri.Segments.Length - 1 [](start = 54, length = 32)

for my education, can you pls. explain why these are the same in existing scenarios?

Reply:
The format should be added as a comment to make this method more readable.

Reply:
Added comments.